src/plots/plots.js

```diff
@@ -1446,6 +1446,29 @@ plots.supplyLayoutGlobalDefaults = function(layoutIn, layoutOut, formatObj) {
 
     var template = layoutIn.template;
     if(Lib.isPlainObject(template)) {
+        if( // geo.visible false should override template.layout.geo.show* - see issue 4482
+            layoutIn.geo &&
+            layoutIn.geo.visible === false &&
+            template.layout &&
+            template.layout.geo
```

Comment:
This is a clever solution! But if we do it this way it will need some extending. `template.layout.geo` will apply to `layout.geo` but also `layout.geo2`, `.geo3`, etc... unless there's also a `template.layout.geo2`, in which case THAT will only apply to `layout.geo2`, nothing before or after.

Oh actually... that means this strategy won't work. If you have `template.layout.geo.showland=true`, no `template.layout.geo2`, but `layout.geo.visible=true` and `layout.geo2.visible=false`, what do you do with the template? It would need to appear `show*=false` to `geo2` but the original setting to `geo`.

So I think the solution is going to have to be at the `supplyDefaults` level, not at the template level... where it seems like we'll need a special coerce variant involving `geoLayoutIn.visible`, `geoLayoutOut._template.visible`, `geoLayoutIn[x]`, and `geoLayoutOut._template[x]`

Unless you want to, in this case, copy `newTemplate.layout.geo` into `newTemplate.layout.geo2` and modify only one of them... but it seems to me given all this extra complexity a `supplyDefaults` solution will be easier.

Reply:
Good call. Fixed by https://github.com/plotly/plotly.js/pull/4483/commits/7e62ce45f14c1a1abdb05c49a631ca4f63439019.